script.js

```diff
@@ -1,4 +1,255 @@
-var main = function (input) {
-  var myOutputValue = 'hello world';
+//GLOBAL VARIABLES
+var gameState = "preGame";
+var playerTurn = 1;
+var roundNumber = 0;
+
+var numberOfDice = 0;
+
+var numberOfPlayers = 0;
+
+//only works up to 4 players - because idk how to execute the loop without creating a fixed number of existing array variables
+var p1DiceCombo = 0;
+var p2DiceCombo = 0;
+var p3DiceCombo = 0;
+var p4DiceCombo = 0;
+
+//player score totals
+var player1Score = 0;
+var player2Score = 0;
+var player3Score = 0;
+var player4Score = 0;
+
+//ROLL DICE FUNCTION
+var rollDice = function () {
+  var randomDecimal = Math.random() * 6;
+  var randomInteger = Math.floor(randomDecimal);
+  var diceNumber = randomInteger + 1;
+  return diceNumber;
+};
+
+var displayNewResultsMessage = function () {
+  myOutputValue =
+    roundCombinationMessage() + overallScoreboardMessage() + leaderMessage();
+  return myOutputValue;
+};
```

Comment:
nice using function to make it more modular making the display of message easier